zipline/data/data_portal.py

```diff
@@ -248,21 +247,23 @@ def get_spot_value(self, asset, column, dt=None):
                 # given a dt to use.
                 minute_offset_to_use = self.cur_data_offset
             else:
-                # dt was passed in, so calculate the offset.
-                # = (390 * number of trading days since 1/2/2002) +
-                #   (index of minute in day)
-
-                # FIXME is this expensive?
-                if not self.env.is_market_hours(dt):
-                    dt = self.env.previous_market_minute(dt)
-
-                given_day = pd.Timestamp(dt.date(), tz='utc')
-                day_index = tradingcalendar.trading_days.searchsorted(
-                    given_day) - INDEX_OF_FIRST_TRADING_DAY
-                minute_index = self.env.market_minutes_for_day(given_day).\
-                    searchsorted(dt)
-
-                minute_offset_to_use = (day_index * 390) + minute_index
+                if dt == self.current_dt:
```

Comment:
@jbredeche this might be a good use-case for encapsulating in a [CachedObject](https://github.com/quantopian/zipline/blob/master/zipline/utils/cache.py#L11)